### PR TITLE
OCPBUGS-37201: Ignore comment symbols when reading the ini file

### PR DESCRIPTION
**Pull request by bertinatto**
CC @openshift/storage

### CODE REVIEW

**Comment by bertinatto** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn-techpreview
/payload-job periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn-techpreview-serial
/payload-job periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn-upi

File: pkg/operator/vspherecontroller/vspherecontroller.go [hunk 1]
```diff
@@ -688,7 +688,8 @@ func (c *VSphereController) applyClusterCSIDriverChange(
 		csiConfigString = strings.ReplaceAll(csiConfigString, pattern, value)
 	}
 
-	csiConfig, err := iniv1.Load([]byte(csiConfigString))
+	// We ignore inline comments when reading the INI file because passwords may contain comment symbos (# and ;)
+	csiConfig, err := iniv1.LoadSources(iniv1.LoadOptions{IgnoreInlineComment: true}, []byte(csiConfigString))
```
Comment (gnufied) [inline pkg/operator/vspherecontroller/vspherecontroller.go:692]: for a uninitiated reader, can you add a comment that describes what this does?

Comment (bertinatto) [reply pkg/operator/vspherecontroller/vspherecontroller.go:692]: Did you see the comment above this line? Or do you mean something else?

Comment (gnufied) [reply pkg/operator/vspherecontroller/vspherecontroller.go:692]: No that was it. I seems to have missed it. You mentioned you wanted some unit test for this stuff before we merge?

Comment (bertinatto) [reply pkg/operator/vspherecontroller/vspherecontroller.go]: Yup, done. PTAL

**Pull request merged**